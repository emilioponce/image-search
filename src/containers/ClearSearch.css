.button {
  border: none;
  font-size: 16px;
  background-color: #61dafb;
  color: #000000;
  padding: 5px 20px;
  margin-left: 5px;
  white-space: nowrap;
  transition: background-color 0.2s ease-out;
  -webkit-transition: background-color 0.2s ease-out;
  -moz-transition: background-color 0.2s ease-out;
}

.button:hover {
  background-color: white;
  cursor: pointer;
}

button:focus {
  outline: 0;
}

button:disabled,
button[disabled] {
  background-color: #cccccc;
  color: #666666;
}
