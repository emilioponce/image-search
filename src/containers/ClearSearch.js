import React from 'react'
import { connect } from 'react-redux'
import { getTotalAmountOfResultsSelector } from '../reducers/search'
import PropTypes from 'prop-types'
import { clearSearch } from '../actions'

import './ClearSearch.css'

const ClearSearch = ({ total, clearSearch }) => {
  const handleClick = () => {
    clearSearch()
  }

  return (
    <button
      type="submit"
      className="button"
      onClick={handleClick}
      disabled={!total}
    >
      Clear search
    </button>
  )
}

// to retrieve the amount of results in order to conditionallly render the button disabled
const mapStateToProps = state => {
  return {
    total: getTotalAmountOfResultsSelector(state)
  }
}

// to dispatch actions to clear the search
const mapDispatchToProps = dispatch => {
  return {
    clearSearch: () => dispatch(clearSearch())
  }
}

ClearSearch.propTypes = {
  total: PropTypes.string,
  clearSearch: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClearSearch)
