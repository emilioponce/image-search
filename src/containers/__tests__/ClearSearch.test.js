import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ClearSearch } from '../ClearSearch'

configure({ adapter: new Adapter() })

const mockClearSearchFn = jest.fn()

const component = <ClearSearch total={'0'} clearSearch={mockClearSearchFn} />

describe('ClearSearch component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(component)
  })

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matches', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call the clear search function once', () => {
    wrapper.find('button').simulate('click', { preventDefault() {} })
    expect(mockClearSearchFn.mock.calls.length).toBe(1)
  })

  it('Should show the button disabled if total is null', () => {
    const component = <ClearSearch total={null} clearSearch={jest.fn()} />
    let wrapper = shallow(component)
    expect(wrapper.find('button').props().disabled).toBe(true)
  })

  it('Should show the button enabled if total has a value', () => {
    const component = <ClearSearch total={'1'} clearSearch={jest.fn()} />
    let wrapper = shallow(component)
    expect(wrapper.find('button').props().disabled).toBe(false)
  })
})
