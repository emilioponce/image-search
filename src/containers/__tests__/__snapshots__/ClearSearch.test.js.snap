// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearSearch component Snapshot matches 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  Clear search
</button>
`;
